### DISCUSSION TITLE
Return static page for bots/guests (via getStaticProps) and dynamic page for users (via getInitialProps) at once.

### DISCUSSION THREAD

**Discussion by nikichv** [vercel/next.js#15684] [Help]
Hello there! I wonder is it possible with next.js to serve two versions of one page for bots and registered users at once.
My use case is simple.
I have main page with a lot of content which is the same for all bots/guests. But if registered user comes in, I need to return his personal main page with a little bit different content on it. So I want to be fast as possible with static content for bots and guests and return them static page from server. And return blank page with loaders for registered users and load data on client-side.
By now I have no good idea with zero overhead how to do this scenario and is this even possible?

Right now I think to do next: make a static page with all content for all users groups. If registered user comes in - immediately turn on "pending" state for whole page and re-load data on client side.

**Comment by davidwilde** [comment]
Hey there @nikichv - did you ever solve this? I have the same use case
